Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1323,167 +1243,130 @@ namespace Babylon
             }
             frameBufferHandle = bgfx::createFrameBuffer(static_cast<uint8_t>(attachments.size()), attachments.data(), true);
         }
+        else
+        {
+            assert(!generateStencilBuffer);
+            frameBufferHandle = bgfx::createFrameBuffer(width, height, format, BGFX_TEXTURE_RT);
+        }
 
         texture->Handle = bgfx::getTexture(frameBufferHandle);
-        return Napi::External<FrameBufferData>::New(info.Env(), m_frameBufferManager.CreateNew(frameBufferHandle, width, height));
+        texture->OwnsHandle = false;
+
+        auto& frameBuffer{m_graphicsImpl.AddFrameBuffer(frameBufferHandle, width, height, false)};
+        return Napi::External<FrameBuffer>::New(info.Env(), &frameBuffer);
     }
 
     void NativeEngine::DeleteFrameBuffer(const Napi::CallbackInfo& info)
     {
-        // Check if this native frame buffer is marked as owned by JavaScript.  It's possible that
-        // we do not recognize it as having been passed back to JS in which case BabylonNative should be
-        // responsible for deleting this frame buffer instead.
-        const auto frameBufferData = info[0].As<Napi::External<FrameBufferData>>().Data();
-        if (frameBufferData->OwnedByJS)
-        {
-            delete frameBufferData;
-        }
+        const auto& frameBuffer{*info[0].As<Napi::External<FrameBuffer>>().Data()};
+        m_graphicsImpl.RemoveFrameBuffer(frameBuffer);
     }
 
     void NativeEngine::BindFrameBuffer(const Napi::CallbackInfo& info)
     {
-        const auto frameBufferData = info[0].As<Napi::External<FrameBufferData>>().Data();
-        m_frameBufferManager.Bind(frameBufferData);
+        auto frameBuffer{info[0].As<Napi::External<FrameBuffer>>().Data()};
+        m_boundFrameBuffer = frameBuffer;
     }
 
     void NativeEngine::UnbindFrameBuffer(const Napi::CallbackInfo& info)
     {
-        const auto frameBufferData = info[0].As<Napi::External<FrameBufferData>>().Data();
-        m_frameBufferManager.Unbind(frameBufferData);
+        const auto frameBuffer{info[0].As<Napi::External<FrameBuffer>>().Data()};
+
+        assert(frameBuffer == m_boundFrameBuffer);
+        UNUSED(frameBuffer);
```

Comment:
```suggestion
        assert(info[0].As<Napi::External<FrameBuffer>>().Data() == m_boundFrameBuffer);
```

Reply:
I agree this will work, but I'm not sure I like it. I'm trying to make the code follow the patterns for getting arguments, but maybe I'm overthinking it.

Reply:
Fair enough, not an important suggestion.